### PR TITLE
Add user's account methods

### PR DESCRIPTION
**Pull request by karolinakuzniewicz**

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-3868 Add rest user's account mutations](https://app.clickup.com/t/2549495/SALEOR-3868)

**Review by krzysztofwolski — APPROVED** [review]

**Review by jwm0 — CHANGES_REQUESTED** [review]

File: test/setup.ts [hunk 1]
```diff
@@ -56,6 +61,32 @@ export const setupRecording = (): Context =>
...
         }
+        if (req.recordingName === "auth api/changes user's password") {
+          // Make auth tests recording to ignore changes in the password variables
+          const json = JSON.parse(body);
+          delete json.variables.newPassword;
+          delete json.variables.oldPassword;
+          return JSON.stringify(json);
+        }
+        if (
+          req.recordingName === "user api/sends request to change user's email"
```
Comment (krzysztofwolski) [inline test/setup.ts:72]: Already in line 49 so this line does not execute at all.

File: test/setup.ts [hunk 2]
```diff
@@ -21,20 +21,17 @@ export const setupRecording = (): Context =>
...
-      url: {
-        hash: false,
-        hostname: false,
-        password: false,
-        pathname: false,
-        port: false,
-        protocol: false,
-        query: false,
-        username: false,
+        exclude: ["host", "content-length"],
```
Comment (krzysztofwolski) [inline test/setup.ts:24]: Why remove authorization header? was it conflicting with tests?

File: test/setup.ts [hunk 3]
```diff
@@ -48,6 +45,14 @@ export const setupRecording = (): Context =>
...
             "auth api/refreshes the auth token",
             "auth api/logout clears user cache",
+            "auth api/verifies if token is valid",
+            "user api/sends request to change user's email",
+            "user api/updates the user's first name",
+            "user api/creates the user's account address",
+            "user api/updates the user's account address",
+            "user api/sets address as a default billing address",
+            "user api/deletes user address",
+            "user api/deletes user account",
```
Comment (krzysztofwolski) [inline test/setup.ts:55]: Such lists doesnt scale well and are area for issues. @jwm0 Ideas how we can refactor it? Remove fields like email and redirect URL from all test matches?

File: test/user.test.ts [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+    lastName: "Test lastname",
+    streetAddress1: "Test street address",
+    city: "Test city",
+    postalCode: "12-345",
+    country: "PL" as CountryCode,
+  };
+
+  beforeEach(async () => {
+    const { server } = context.polly;
+    server.any().on("beforePersist", (_, recording) => {
```
Comment (krzysztofwolski) [inline test/user.test.ts:21]: WDYT bout extract this part from both files and don't duplicate?

File: test/user.test.ts [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+      type: "BILLING",
+    });
+    expect(
+      data?.accountSetDefaultAddress?.user?.defaultBillingAddress?.id
+    ).toBe(addressId);
+    expect(data?.accountSetDefaultAddress?.errors).toHaveLength(0);
+
+    // Set back default testing address
+    const { data: revertedData } = await saleor.user.setAccountDefaultAddress({
+      id: oldAddress.id,
```
Comment (krzysztofwolski) [inline test/user.test.ts:132]: Getting `TypeError: Cannot read property 'id' of null` from this line.
BTW why are we need to revert old address?

Comment (karolinakuzniewicz) [reply test/user.test.ts:132]: No `TypeError` by me, the test was recorded fine. I wanted to go back to the current test address, not to set the dummy address.

Comment (krzysztofwolski) [reply test/user.test.ts:132]: I got it why it fails - shared account no longer has any address

File: test/user.test.ts [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+      type: "BILLING",
+    });
+    expect(
+      revertedData?.accountSetDefaultAddress?.user?.defaultBillingAddress?.id
+    ).toBe(oldAddress.id);
+    expect(revertedData?.accountSetDefaultAddress?.errors).toHaveLength(0);
+  });
+
+  it("deletes user address", async () => {
```
Comment (krzysztofwolski) [inline test/user.test.ts:141]: If run on staging with a shared test account, then it's not guaranteed this test will pass - there can be no address.

File: test/setup.ts (hunk 2 above)
Comment (karolinakuzniewicz) [reply test/setup.ts:24]: I was checking the failing tests and forgot to add it back.

File: test/user.test.ts (hunk 4 above)
Comment (karolinakuzniewicz) [reply test/user.test.ts:21]: good catch

File: test/user.test.ts (hunk 5 above)
Comment (karolinakuzniewicz) [reply test/user.test.ts:132]: should we worry about this? We already have recorded a test recording.

File: test/user.test.ts (hunk 6 above)
Comment (krzysztofwolski) [reply test/user.test.ts:122]: If you want to be sure that some address exists - create new one

File: test/user.test.ts (hunk 5 above)
Comment (krzysztofwolski) [reply test/user.test.ts:113]: We should - tests must be as independent as possible. 
The first record is not an issue in most cases, problems arise when the next person changes "one little thing" and has to refresh recordings. And then has to fight with some assumed server state. 
Solution: you can create a new user in the test setup and run mutations using its account. 
Also in this test you can create 2 address and check if switching works as intended without depending on some server state.

File: test/setup.ts (hunk 3 above)
Comment (jwm0) [reply test/setup.ts:76]: I agree, let's just simply maintain a blacklist of all variables that need to be deleted before creating the recording and use it everywhere:
```
const variablesBlacklist = [
  "email",
  "password",
  "redirectUrl",
  "newPassword",
  "oldPassword",
  "newEmail"
];

body(body, req): string {
  const json = JSON.parse(body);

  variablesBlacklist.forEach((key) => {
    delete obj.variables[key];
  })

  return JSON.stringify(json);
}
```

**Review by jwm0 — CHANGES_REQUESTED** [review]

**Review by jwm0 — APPROVED** [review]

**Pull request merged**